azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/auth/oauth/AzureLoginHelper.java

```diff
@@ -0,0 +1,126 @@
+        final WebAppServer webAppServer = new WebAppServer();
+        final int port = webAppServer.getPort();
+        final String baseURL = baseURL(env); 
+        final String redirectUrl = "http://localhost:" + port;
+        final String authorizationUrl = String.format(
+                "%s/oauth2/authorize?client_id=%s&response_type=code" +
+                "&redirect_uri=%s&&prompt=select_account&resource=%s",
+                baseURL, CLIENT_ID, redirectUrl, env.managementEndpoint());
+        try { 
+            Desktop.getDesktop().browse(new URL(authorizationUrl).toURI());
+            System.out.println(AUTH_WITH_OAUTH);
+        } catch (Exception ex) {
+            // the browse is not available
+            return null;
+        }
+        ExecutorService executorService = null;
+        try {            
+            webAppServer.start();
+            executorService = Executors.newSingleThreadExecutor();
+            final AuthenticationContext authenticationContext = new AuthenticationContext(baseURL, true, 
+                    executorService);
+            
+            final AuthenticationResult result = authenticationContext.acquireTokenByAuthorizationCode(
+                    webAppServer.getResult(), env.managementEndpoint(), CLIENT_ID,
+                    new URI(redirectUrl), null).get();
+            
+            return new AzureCredential(env, result);
+        } catch (Exception e) {
+            throw new IOException(e.getMessage());
+        } finally {
+            webAppServer.stop();
+            executorService.shutdown();
+        }
+    }
+    
+    public static AzureCredential deviceLogin(AzureEnvironment env) throws IOException {
+        ExecutorService executorService = null;
+        try {
+            System.out.println(AUTH_WITH_DEVICE_LOGIN);
+            executorService = Executors.newSingleThreadExecutor();
+            final AuthenticationContext authenticationContext = 
+                    new AuthenticationContext(baseURL(env), true, executorService);
+            final DeviceCode deviceCode = authenticationContext.acquireDeviceCode(CLIENT_ID, 
+                    env.activeDirectoryResourceId(), null).get();
+            
+            System.out.println(String.format(LOGIN_MESSAGE_TEMPLATE,
+                    deviceCode.getVerificationUrl(), deviceCode.getUserCode()));
+            try {
+                // disable log4j of AuthenticationContext, otherwise the pending user authorization log
+                // will be print every second.
+                // see https://github.com/AzureAD/azure-activedirectory-library-for-java/issues/246
+                final Object logger = LoggerFactory.getLogger(AuthenticationContext.class);  
+                final Field levelField = logger.getClass().getSuperclass().getDeclaredField("currentLogLevel");
+                levelField.setAccessible(true);
+                levelField.set(logger, LocationAwareLogger.ERROR_INT + 1); 
```

Comment:
Shall we turn back the log level after device login?

Reply:
Doesn't need since AuthenticationContext will be only used here